ui/desktop/src/main.ts

```diff
@@ -944,8 +944,42 @@ const buildRecentFilesMenu = () => {
 const openDirectoryDialog = async (
   replaceWindow: boolean = false
 ): Promise<OpenDialogReturnValue> => {
+  // Get the current working directory from the focused window
+  let defaultPath: string | undefined;
+  const currentWindow = BrowserWindow.getFocusedWindow();
+
+  if (currentWindow) {
+    try {
+      const currentWorkingDir = await currentWindow.webContents.executeJavaScript(
+        `window.appConfig ? window.appConfig.get('GOOSE_WORKING_DIR') : null`
+      );
```

Comment:
hmm, what's happening here? where does this working dir come from? is that what the server thinks it is?